### DISCUSSION TITLE
Astro builds URL with trailing slash, even when configured not to

### DISCUSSION THREAD

**Discussion by betarepos** [arthelokyo/astrowind#145] [General]
Hey guys, 

maybe the wrong place, but it wanted to try it anyway. I configured the project to not use trailing slashes, but when building the "oG URLs" (<meta content="https://example.com/" property="og:url"/>" will be built with trailing slash, which overall leads to 308 redirects to not trailing slash URLs. Where does this come from and where can i configure it?
i configured the config.mjs and vercel.json to trailingslash: false

Can it be the getCanonical function?

**Comment by prototypa** [comment]
Hello @betarepos 

Thanks for bringing this to the table. It's very important.

It seems that in one of the latest _Astro_ updates, something changed and now the problem only occurs in the build, I have seen that the dev environment generates the URLs correctly. I am reviewing in detail what it could be. I will write to you as soon as I have any new information.

Thank you.

**Comment by prototypa** [comment]
Hi @betarepos 

It's resolved now with a small tweak: https://github.com/onwidget/astrowind/commit/a608eff456f7b67a527a3768585d4f2a4990b7d6

I made some adjustments to `getCanonical`. Before, the path received from _Astro_ in [MetaTags.astro](https://github.com/onwidget/astrowind/blob/main/src/components/common/MetaTags.astro#L34) (Astro.url.pathname) was according to the **trailingSlash** configuration, currently in the build that does not happen.

I am going to investigate more in the latest versions of _Astro_ to check if it is a mistake or a decision.

Thank you again.

**Reply by betarepos** [reply]
Awesome, looking good :)
